### ISSUE TITLE
Fill out the partial include path automatically

### ISSUE THREAD

**Issue by msuchane** [redhat-documentation/newdoc#16]
See <https://github.com/redhat-documentation/tools/issues/33>.

**Comment by msuchane** [comment]
One possible algorithm to determine the partial path:

1. Take the current working directory (CWD).
2. Browse the CWD from the end.
3. When you encounter a `modules` or `assemblies` directory (in that order of priority), stop and discard the rest of the CWD path, from the start until this point.
4. Use the result, which starts with `modules/` or `assemblies/` and can continue with more nested directories.
5. If the CWD path doesn't include `modules` or `assemblies` anywhere, fall back on the `<path>` placeholder that `newdoc` uses now.

For more reliability, check if the `modules` or `assemblies` directory is located in a Git repository. If it isn't, fall back on the placeholder.

**Comment by msuchane** [comment]
Another problem we need to solve is how to deal with the mass-created module files, generated using the `--include-in` option for an assembly. With current newdoc, these modules are placed in the same file as the assembly, so the CWD path is obviously incorrect for them.

And now I realize that in the algorithm I described in https://github.com/redhat-documentation/newdoc/issues/16#issuecomment-698990395, `newdoc` shouldn't look for `modules` or `assemblies`, but rather it should search for `modules` if the file is a module and `assemblies` if the file is an assembly.

**Comment by rkratky** [comment]
@mrksu, how about prompting the user to:

* Accept the heuristically determined path (default)
* Enter the correct path manually
* Reject and continue with placeholder

All of the above only if the feature would be triggered by a non-default option.

In case the `--include-in` option is used together with this new option, the prompt would offer (and the `include` statements would be generated accordingly):

* Create all modules in the same directory (default)
* Enter the desired path manually (relative to CWD)

**Comment by msuchane** [comment]
Hi @yzimmerm and @rkratky, I've added an option in `newdoc` that tries to fill out the include path automatically: `--detect-directory` (`-D`).

It doesn't prompt you for a manual input like @rkratky suggested. It either finds a suitable path or falls back on `<path>`. But I'll keep the suggestion to consider it in the future, if people are interested.

When you have a moment, could you please help me test the option? Just update `newdoc` to the latest version from the [current repository](https://copr.fedorainfracloud.org/coprs/mareksu/newdoc-rs/) and try generating files with the `--detect-directory` flag.

For now, this is an optional feature. If the testing goes well, I could eventually make this behavior the default.

**Comment by msuchane** [comment]
I've just released `newdoc` 2.7.0, which tries to guess the include path by default. In other words, the `--detect-directory` option that I described earlier is now enabled by default, and setting the option on the command line has no additional effect.

This is now the output that `newdoc` displays:

```
modules/filesystems-and-storage]$ newdoc -c "A test of the include path"
[INFO] ‣ File generated: ./con_a-test-of-the-include-path.adoc
[INFO]   include::modules/filesystems-and-storage/con_a-test-of-the-include-path.adoc[leveloffset=+1]
```

I think that closes this issue.

**Comment by yzimmerm** [comment]
Superb!

Yehuda Zimmerman, RHCSA

Technical Writer III

Red Hat EMEA <https://www.redhat.com/>

***@***.***
M: +972-50-942-1156     IM: yzimmerm
<https://red.ht/sig>
